File: classify-rest/src/main/java/io/ermdev/alice/entity/Teacher.java
package io.ermdev.alice.entity;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Table(name = "tblteacher")
@Entity
public class Teacher {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String firstName;
    private String lastName;

    @OneToMany(mappedBy = "teacher")
    private List<Class> classes = new ArrayList<>();

    public Teacher(){}

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public List<Class> getClasses() {
        return classes;
    }

    public void setClasses(List<Class> classes) {
        this.classes = classes;
    }
}

File: classify-rest/src/main/java/io/ermdev/classify/data/entity/Teacher.kt
package io.ermdev.classify.data.entity

import javax.persistence.*

@Entity
@Table(name = "tbl_teacher")
class Teacher(
        @Id
        @GeneratedValue(strategy = GenerationType.AUTO)
        var id: Long = 0,

        var name: String = "",

        var email: String = "",

        @OneToOne
        @JoinColumn(name = "user_id")
        var user: User? = null)